### PR TITLE
MP-5113  Data Transformer Lens

### PR DESCRIPTION
**Pull request by danilmoro3**
Ticket: https://spryker.atlassian.net/browse/MP-5113

### CODE REVIEW

File: docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        },
+      },
+    },
+  }"
+></spy-table>
+```
+
+## Interfaces
+
+`path` - the name of the object property, whose value needs to be transformed via configuration. May contain nested properties separated by dots, just like in a Javascript language.<br>
```
Comment (gund) [inline docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md:59]:
```suggestion
`path` - the name of the object property, whose value needs to be transformed via configuration. May contain nested properties separated by dots, just like in a Javascript language.
```

File: docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+This document provides details about the Data Transformer Lens service in the Components Library.
+
+## Overview
+
+Data Transformer Lens is an Angular Service that updates the nested object by path using another Data Transformer set up with a configuration object.
+In the example below date in `col2` will be transformed.
+
+```ts
+<spy-table
+  [config]="{
```
Comment (gund) [inline docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md:17]: Please do not use table in examples as it's more complicated.
Better to use `spy-select` with just the datasource and transformer example.

File: docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    columns: [
+      { id: 'col1', title: 'Column #1' },
+      { id: 'col2', title: 'Column #2' },
+    ],
+    dataSource: {
+      type: 'table.inline',
```
Comment (gund) [inline docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md:23]: Please use simple `inline` datasource instead.

**Review by gund — CHANGES_REQUESTED** [review]

File: docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md (hunk 1 above)
Comment (danilmoro3) [reply docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md:59]: How to move text to the next line in this case?

Comment (danilmoro3) [reply docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md:59]: fyi: this tag visible only in markup

Comment (gund) [reply docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md:50]: Either double space in the end of the line or use two new lines.

**Review by gund — CHANGES_REQUESTED** [review]

File: docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md [hunk 4]
```diff
@@ -10,48 +10,39 @@ This document provides details about the Data Transformer Lens service in the Co
...
+    transform: {
+      type: 'lens',
+      path: 'date',
+      transformer: {
+        type: 'array-map',
+        mapItems: {
+          type: 'object-map',
+          mapProps: {
+            date: {
+              type: 'date-parse',
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply docs/marketplace/dev/front-end/ui-components-library/data-transformers/lens.md]: Fixed

**Review by gund — APPROVED** [review]

**Pull request merged**